### PR TITLE
Make sure static mdc from withMdc set during actor startup 

### PR DESCRIPTION
**Pull request by johanandren**
References #29893

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Makes sense to me :+1:

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala [hunk 1]
```diff
@@ -348,7 +348,8 @@ class ActorLoggingSpec extends ScalaTestWithActorTestKit("""
         .info("Starting")
         // not counting for example "akkaSource", but it shouldn't have any other entries
         .withCustom(logEvent =>
-          logEvent.mdc.keysIterator.forall(entry => entry.startsWith("akka") || entry == "sourceActorSystem"))
+          logEvent.mdc.keysIterator.forall(entry =>
+            entry.startsWith("akka") || entry == "sourceActorSystem" || entry == "static"))
```
Comment (patriknw) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala:352]: this doesn't test that `"static" -> "1"` is included in MDC of the "starting" logging

**Review by patriknw — COMMENTED** [review]
good, but the test can maybe be improved

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala]: Done!

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**